### PR TITLE
Add `--multiplicity-weighted-elbo` commandline flag 

### PR DESCRIPTION
**Pull request by kmdalton**
This PR adds a new flag to the CLI which controls how the ELBO is calculated. 

```
--multiplicity-weighted-elbo
                      Reweight the kl_div term by multiplicity. This may perform better in some cases.
```
When this flag is set, there are two major changes in the loss function. 
 - The log likelihood and kl_div terms are put on equal scales. This simply amounts to calling `tf.reduce_mean` instead of the historical `tf.reduce_sum`. 
 - The merger expands each of the `kl_div` terms before summing. This accounts for differences in multiplicity between reflections in the data set. 
 
 Why would I want to do this? Anecdotally, this strategy seems to produce results that are more in line with classically scaled data. So, setting this flag may lead to output that is more agreeable to downstream processing. Furthermore, this method is more readily phrased as a batched objective suitable for stochastic training. In the future, variational merging codes are most likely going to use a formulation of the ELBO like one controlled by this flag. 
 
 My inclination is that this is a superior model to the original `careless` formulation. This mode may become the new default int he future. As it stands, this claim still needs extensive validation.

### CODE REVIEW

**Review by JBGreisman — APPROVED** [review]
All looks good to me
